dask/dataframe/io/csv.py

```diff
@@ -915,6 +915,8 @@ def to_csv(
         values = [value]
         files = [first_file]
     else:
+        if os.path.isdir(filename) and "*" not in filename:
```

Comment:
A suggestion: most Dask I/O functions use `fsspec` as an abstract interface to file-system-like storage backends. This is what allows us to easily write to things like cloud storage using the same APIs as local filesystems.

`os.path.isdir` will *only* work for the local filesystem. Instead, I suggest using something like `fsspec.get_fs_token_paths` or `fsspec.url_to_fs`:

```python
fs = fsspec.url_to_fs(filename)
if fs.isdir(filename) and "*" not in filename:
    filename = fs.sep.join(filename, "part.*.csv")
```